From 2bd474a618a74352d02976d7207022888863e58b Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Sun, 8 Mar 2026 08:59:19 +0800
Subject: [PATCH 126/133] std.zig.system.loongarch: remove PRID-based
 detections
X-Developer-Signature: v=1; a=openpgp-sha256; l=946; i=xtex@astrafall.org;
 h=from:subject; bh=UgFfwot2/6wYctPjdmX0b4cO5HzZsD9AKA+aOi/2kM8=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78vg7Nu4+xhsZdOGw3nz3H5sNE7adCbzAL3+ly+Htl
 q8P5/WndExkYRDjYrAUU2QpMmzwZtVJ5xddVi4LM4eVCWSItEgDAwMDAwsDX25iXqmRjpGeqbah
 nqGhjpGOEQMXpwBMtfcihn/WKat3HkvbxmClHbzSKWiuThvvHVWGxGc/jZqazpe5SSUyMpz7qDz
 L+rv4jNNqszwXF157uMO/6LviCZXvlYL1y3yZNbgA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Remove it because:
- CPUCFG-based detections are already support
- Some CPU features might not be supported by OS kernels

Change-Id: I03e03bca146fbdad1b7dad758ae59fef614901d0
Signed-off-by: xtex <xtex@astrafall.org>
---
 lib/std/zig/system/loongarch.zig | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/lib/std/zig/system/loongarch.zig b/lib/std/zig/system/loongarch.zig
index 7a2c62279131..2186ce6c76a2 100644
--- a/lib/std/zig/system/loongarch.zig
+++ b/lib/std/zig/system/loongarch.zig
@@ -21,11 +21,7 @@ pub fn detectNativeCpuAndFeatures(
 
     var cpu: std.Target.Cpu = .{
         .arch = arch,
-        .model = switch (cpucfg(0) & 0xf000) {
-            else => return null,
-            0xc000 => &std.Target.loongarch.cpu.la464,
-            0xd000 => &std.Target.loongarch.cpu.la664,
-        },
+        .model = .generic(arch),
         .features = .empty,
     };
 
-- 
2.52.0

